crates/bevy_ecs/src/schedule_v3/executor/multi_threaded.rs

```diff
@@ -1,14 +1,18 @@
-use bevy_tasks::{ComputeTaskPool, Scope, TaskPool};
+use std::panic::AssertUnwindSafe;
```

Comment:
One alternative that wasn't listed that wouldn't affect perf: we could add a `scope.spawn_last()` function that calls `await` on those tasks only after normal tasks have finished. It would mean adding another ConcurrentQueue to Scope which would increase the size of Scope, adds another bleh transmute into the equation, and would allocate unless we use ConcurrentQueue::bounded(1)

Reply:
I'm clearly missing something because this doesn't work as I expected when implement I spawn_last with a `ConcurrentQueue` and use `scope.spawn_last(executor)`:

```rust
let get_results = async {
    let mut results =
        Vec::with_capacity(spawned_ref.len() + spawned_last_ref.len());
    let mut index = 0;
    let mut task_failed = false;
    while let Ok(task) = spawned_ref.pop() {
        if task_failed {
            task.cancel().await;
        } else {
            if let Some(result) = task.await {
                unsafe { *results.get_unchecked_mut(index) = result };
            } else {
                task_failed = true;
            }
        }
        index += 1;
    }
    // this detects _no_ failed tasks when a system panics. index is 0
    // prints "false 0"
    println!("{task_failed} {index}");

    while let Ok(task) = spawned_last_ref.pop() {
        if task_failed {
            task.cancel().await;
        } else {
            if let Some(result) = task.await {
                unsafe { *results.get_unchecked_mut(index) = result };
            } else {
                task_failed = true;
            }
        }
        index += 1;
    }

    // this does not run because we hit the deadlock above for the executor task
    println!("{task_failed} {index}");
    
    if task_failed {
        panic!("At least one task failed");
    }
    unsafe { results.set_len(index) };
    results
};
```

Reply:
Apparently this scope isn't registering the system function at all?

Reply:
For the curious: https://github.com/cart/bevy/tree/spawn_last

Reply:
I think I answered my own question on discord:
![image](https://user-images.githubusercontent.com/2694663/216526563-1768bb57-9377-4360-85f2-6ade37112fbf.png)

Reply:
I don't think spawn_last is an option here.

Reply:
If you're using the stage less branch, this  looks very wrong as the first time you enter get_results the only task might be the executor task.  When that task runs it then spawns the system tasks. So get_results needs to be able to handle more tasks spawning while it runs.